src/metkit/api/metkit_c.h

```diff
@@ -0,0 +1,238 @@
+
+struct metkit_paramiterator_t;
+/** Iterator for iterating over parameters in Request */
+typedef struct metkit_paramiterator_t metkit_paramiterator_t;
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * ERROR HANDLING
+ * -------------- */
+
+typedef enum metkit_error_values_t
+{
+    METKIT_SUCCESS            = 0, /* Operation succeded. */
+    METKIT_ITERATION_COMPLETE = 1, /* All elements have been returned */
+    METKIT_ERROR              = 2, /* Operation failed. */
+    METKIT_ERROR_UNKNOWN      = 3, /* Failed with an unknown error. */
+    METKIT_ERROR_USER         = 4, /* Failed with an user error. */
+    METKIT_ERROR_ASSERT       = 5  /* Failed with an assert() */
+} metkit_error_enum_t;
+
+const char* metkit_get_error_string(int err);
+
+/* -----------------------------------------------------------------------------
+ * HELPERS
+ * ------- */
+
+/**
+ * @brief Set MetKit version.
+ *
+ * @param version Version string
+ * @return int Error code
+ */
+int metkit_version(const char** version);
+
+/**
+ * @brief Set MetKit git sha1 version.
+ *
+ * @param sha1 SHA1 version string
+ * @return int Error code
+ */
+int metkit_vcs_version(const char** sha1);
+
+/**
+ * @brief Initialise Main() context.
+ *
+ * @note This is ONLY required when Main() is NOT initialised, such as loading
+ * the MetKit as shared library in Python.
+ * @return int Error code
+ */
+int metkit_initialise();
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * PARSING
+ * --- */
+
+/**
+ * Parse MARS requests into RequestIterator of Request instances. Resulting RequestIterator
+ * must be deallocated with metkit_free_requestiterator
+ * @param str MARS requests
+ * @param requests Allocates RequestIterator object
+ * @return int Error code
+ */
+int metkit_parse_marsrequest(const char* str, metkit_requestiterator_t** requests, bool strict = false);
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * REQUEST
+ * --- */
+
+/** Allocates new Request object. Must be deallocated with mekit_free_request
+ * @param request new Request instance
+ * @return int Error code
+ */
+int metkit_new_marsrequest(metkit_marsrequest_t** request);
+
+/** Deallocates Request object and associated resources.
+ * @param request Request instance
+ * @return int Error code
+ */
+int metkit_free_marsrequest(const metkit_marsrequest_t* request);
+
+/** Add parameter and values to request
+ * @param request Request instance
+ * @param param parameter name
+ * @param values array of values for parameter
+ * @param numValues number of values
+ * @return int Error code
+ */
+int metkit_marsrequest_add(metkit_marsrequest_t* request, const char* param, const char* values[], int numValues);
+
+/** Add parameter and values to request
```

Comment:
Would be nice to have some documentation on why the 1 appears in the function name. Or even better: renaming to function to reflect what is happening.